drivers/serial/uart_stm32.c

```diff
@@ -751,6 +1290,16 @@ static void uart_stm32_irq_config_func_##index(struct device *dev)	\
 #define STM32_UART_IRQ_HANDLER(index)
 #endif
 
+#ifdef CONFIG_UART_ASYNC_API
+#define STM32_UART_DMA_DATA(index)					\
+	UTIL_AND(DT_INST_DMAS_HAS_NAME(index, rx),			\
```

Comment:
Can you check https://github.com/zephyrproject-rtos/zephyr/pull/25286 ?
It will allow more flexible use of channels

Reply:
Thanks for the comments! I'll refer to that changes and address your comments.

Reply:
Actually, I'm a bit confused. My current implementation does support disabling DMA since the STM32_UART_DMA_DATA() will be nothing if `CONFIG_UART_ASYNC_API` is not defined. What do you mean allowing more flexible use of channels? I've checked #25286 and it seems to do the similar implementation to mine.

Reply:
Aim of #25286 is to allow use and definition (in dts nodes) of only one DMA channel Tx or Rx.
With current implementation, there is one case where this won't compile.

Reply:
I see, will check it later.